manager/orchestrator/service.go

```diff
@@ -44,8 +46,11 @@ func DeleteServiceTasks(ctx context.Context, s *store.MemoryStore, service *api.
 	err = s.Batch(func(batch *store.Batch) error {
 		for _, t := range tasks {
 			err := batch.Update(func(tx store.Tx) error {
-				if err := store.DeleteTask(tx, t.ID); err != nil {
-					log.G(ctx).WithError(err).Errorf("failed to delete task")
+				// update desired state to REMOVE
+				t.DesiredState = api.TaskStateRemove
+
+				if err := store.UpdateTask(tx, t); err != nil {
+					log.G(ctx).WithError(err).Errorf("failed to update task desired state to REMOVE")
 				}
 				return nil
 			})
```

Comment:
The log statement here https://github.com/docker/swarmkit/pull/2446/files#diff-93afa918180cbe7ddd6db3b78a89fb8aR64 doesn't seem accurate. Its the failure of task update txn.